internal/handlers/pg/pgdb/transactions.go

```diff
@@ -54,36 +54,58 @@ func (e *transactionConflictError) Error() string {
 // so the caller needs to use errors.Is to check the error,
 // for example, errors.Is(err, ErrSchemaNotExist).
 func (pgPool *Pool) InTransaction(ctx context.Context, f func(pgx.Tx) error) (err error) {
+	var keepTx pgx.Tx
+
+	err = pgPool.InTransactionKeep(ctx, func(tx pgx.Tx) error {
+		keepTx = tx
+		return f(tx)
+	})
+	if err != nil {
+		err = lazyerrors.Error(err)
+		return
+	}
+
+	err = keepTx.Commit(ctx)
+	if err != nil {
+		err = lazyerrors.Error(err)
+		_ = keepTx.Rollback(ctx)
```

Comment:
On a rare occasion where rollback on commit fails, do we want to log the error here too?

Reply:
Commit and rollback are already logged by our pgx logging configuration

Reply:
I had no idea thanks :+1: